### PR TITLE
Fix the sync issue with non-java files, resolvePath for non-java file…

### PR DESCRIPTION
**Pull request by andxu**
… should not return an empty path

Signed-off-by: andxu <andxu@microsoft.com>

### CODE REVIEW

**Comment by andxu** [comment]
https://github.com/Microsoft/vscode-java-dependency/issues/79

File: jdtls.ext/com.microsoft.jdtls.ext.core/src/com/microsoft/jdtls/ext/core/PackageCommand.java [hunk 1]
```diff
@@ -164,11 +160,88 @@
...
+            if (resource != null) {
+                IResource parent = resource.getParent();
+                IJavaElement parentJavaElement = JavaCore.create(parent);
+                if (parent instanceof IFolder && parentJavaElement instanceof IPackageFragment) {
+                    IPackageFragment packageFragment = (IPackageFragment) parentJavaElement;
+                    String packageName = packageFragment.isDefaultPackage() ? DEFAULT_PACKAGE_DISPLAYNAME : packageFragment.getElementName();
+                    PackageNode packageNode = new PackageNode(packageName, packageFragment.getPath().toPortableString(), NodeKind.PACKAGE);
+                    IPackageFragmentRoot pkgRoot = (IPackageFragmentRoot) packageFragment.getParent();
+                    PackageNode rootNode = null;
+                    if (typeRoot instanceof IClassFile) {
```
Comment (yaohaizh) [inline jdtls.ext/com.microsoft.jdtls.ext.core/src/com/microsoft/jdtls/ext/core/PackageCommand.java]: typeRoot is already null...

File: jdtls.ext/com.microsoft.jdtls.ext.core/src/com/microsoft/jdtls/ext/core/PackageCommand.java [hunk 2]
```diff
@@ -164,11 +160,88 @@
...
+                        rootNode = new PackageRootNode(pkgRoot.getElementName(), pkgRoot.getPath().toPortableString(), NodeKind.PACKAGEROOT, pkgRoot.getKind());
+                    } else {
+                        rootNode = new PackageRootNode(
+                                ExtUtils.removeProjectSegment(packageFragment.getJavaProject().getElementName(), pkgRoot.getPath()).toPortableString(),
+                                pkgRoot.getPath().toPortableString(), NodeKind.PACKAGEROOT, pkgRoot.getKind());
+                    }
+
+                    IProject proj = parentJavaElement.getJavaProject().getProject();
```
Comment (yaohaizh) [inline jdtls.ext/com.microsoft.jdtls.ext.core/src/com/microsoft/jdtls/ext/core/PackageCommand.java]: This part should be duplicate code

File: jdtls.ext/com.microsoft.jdtls.ext.core/src/com/microsoft/jdtls/ext/core/PackageCommand.java [hunk 3]
```diff
@@ -164,11 +160,88 @@
...
+                } else {
+                    return getParentListToProject(resource);
+                }
+            }
         }
 
         return result;
     }
 
+
```
Comment (yaohaizh) [inline jdtls.ext/com.microsoft.jdtls.ext.core/src/com/microsoft/jdtls/ext/core/PackageCommand.java]: Misleading method name. Should be ancestor list

File: jdtls.ext/com.microsoft.jdtls.ext.core/src/com/microsoft/jdtls/ext/core/PackageCommand.java [hunk 4]
```diff
@@ -164,11 +160,88 @@
...
+    private static List<PackageNode> getParentListToProject(IResource element) throws JavaModelException {
+        List<PackageNode> nodeList = new ArrayList<>();
+        while (element != null) {
```
Comment (yaohaizh) [inline jdtls.ext/com.microsoft.jdtls.ext.core/src/com/microsoft/jdtls/ext/core/PackageCommand.java]: This part have lots of duplicate code with other parts in this file

**Comment by andxu** [comment]
using https://github.com/Microsoft/vscode-java-dependency/pull/131 to replace this one

**Pull request closed without merging**